server.js

```diff
@@ -1,32 +1,51 @@
 import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
+import booksData from './data/books.json'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
+const ERROR_BOOKS_NOT_FOUND = {error: "Sorry, we couldn't find any books matching your search"}
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+// Routes
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Welcome to Books API, type in /books for an API of all the books we have in the library')
 })
 
+app.get('/books', (req, res) => {
+  const {authors, title} = req.query;
+  if (authors) {
+    const filteredAuthors = booksData.filter(item => item.authors.toLowerCase().includes(authors.toLowerCase()))
+    filteredAuthors.length === 0 ? res.status(404).json(ERROR_BOOKS_NOT_FOUND) : res.json(filteredAuthors);
+  } else if (title)  {
+    const filteredTitles = booksData.filter(item => item.title.toString().toLowerCase().includes(title.toString().toLowerCase()))
+    filteredTitles.length === 0 ? res.status(404).json(ERROR_BOOKS_NOT_FOUND) : res.json(filteredTitles);
+  }
+  else { 
+    res.json(booksData)
+  }
+})
+
+app.get('/books/top-rated', (req, res) => {
+  const sortedOnRating = booksData.sort((a, b) => b.average_rating - a.average_rating)
+  const topRated = sortedOnRating.slice(0,20)
+  res.json(topRated)
+})
```

Comment:
Wouldn't it be nice if this too were a query under the `/books` endpoint? 😇 Then the user can choose whether to sort it in ascending or descending order
`/books?sort=descending`